packages/web3-core/types/index.d.ts

```diff
@@ -0,0 +1,604 @@
+
+    create(numberOfAccounts: number, entropy?: string): WalletBase;
+
+    add(account: string | AddAccount): AddedAccount;
+
+    remove(account: string | number): boolean;
+
+    clear(): WalletBase;
+
+    encrypt(password: string): EncryptedKeystoreV3Json[];
+
+    decrypt(
+        keystoreArray: EncryptedKeystoreV3Json[],
+        password: string
+    ): WalletBase;
+
+    save(password: string, keyName?: string): boolean;
+
+    load(password: string, keyName?: string): WalletBase;
+}
+
+export interface AddAccount {
+    address: string;
+    privateKey: string;
+}
+
+export interface AddedAccount extends Account {
+    index: number;
+}
+
+export interface Account {
+    address: string;
+    privateKey: string;
+    signTransaction: (
+        transactionConfig: TransactionConfig,
+        callback?: (signTransaction: SignedTransaction) => void
+    ) => Promise<SignedTransaction>;
+    sign: (data: string) => Sign;
+    encrypt: (password: string) => EncryptedKeystoreV3Json;
+}
+
+export interface EncryptedKeystoreV3Json {
+    version: number;
+    id: string;
+    address: string;
+    crypto: {
+        ciphertext: string;
+        cipherparams: { iv: string };
+        cipher: string;
+        kdf: string;
+        kdfparams: {
+            dklen: number;
+            salt: string;
+            n: number;
+            r: number;
+            p: number;
+        };
+        mac: string;
+    };
+}
+
+export interface Sign extends SignedTransaction {
+    message: string;
+    signature: string;
+}
+
+export interface SignatureObject {
+    messageHash: string;
+    r: string;
+    s: string;
+    v: string;
+}
+
+// put all the `web3-provider` typings in here so we can get to them everywhere as this module does not exist in 1.x
+
+export class BatchRequest {
+    constructor(moduleInstance: AbstractWeb3Module);
+
+    add(method: AbstractMethod): void;
+
+    execute(): Promise<BatchError | BatchResponse>;
+}
+
+export interface BatchError {
+    errors: BatchErrorItem[];
+    response: any[];
+}
+
+export interface BatchErrorItem {
+    error: Error;
+    method: AbstractMethod;
+}
+
+export interface BatchResponse {
+    methods: AbstractMethod[];
+    response: any[];
+}
+
+export class ProviderDetector {
+    static detect(): provider;
+}
+
+export class ProvidersModuleFactory {
+    createBatchRequest(moduleInstance: AbstractWeb3Module): BatchRequest;
+
+    createProviderResolver(): ProviderResolver;
+
+    createHttpProvider(url: string): HttpProvider;
+
+    createWebsocketProvider(url: string): WebsocketProvider;
+
+    createIpcProvider(path: string, net: net.Server): IpcProvider;
+
+    createWeb3EthereumProvider(connection: any): Web3EthereumProvider;
+}
+
+export class HttpProvider {
+    constructor(host: string, options?: HttpProviderOptions);
+
+    host: string;
+    connected: boolean;
+
+    supportsSubscriptions(): boolean;
+
+    send(method: string, parameters: any[]): Promise<any>;
```

Comment:
The provider interface isn't correct. The 1.0 providers are implemented with callbacks and different parameters